nomenclature/validation_schemas/variable_schema.yaml

```diff
@@ -0,0 +1,26 @@
+$schema: 'https://json-schema.org/draft/2020-12/schema'
+title: VariableCodeList
+description: >
+  This schema is used for the validation of yaml variable files.
+type: array
+items:
+  $ref: '#/definitions/Variable'
+
+definitions:
+
+  Variable:
+    type: object
+    patternProperties:
+      # The key of this dictionary is the variable name (aka code)
+      ^.+$:
+        type: object
+        # The lower-level dictionary are the attributes
+        properties:
+          unit:
+            type: [ string, "null" ]
+        required:
+          - unit
+        additionalProperties:
+          type: [ string, number, "null" ]
+    minProperties: 1
+    maxProperties: 1
```

Comment:
Is there a `additionalProperties: false` missing to force evaluation against the pattern property?